### DISCUSSION TITLE
Template based on project export

### DISCUSSION THREAD

**Discussion by jwijffels** [inception-project/inception#5153] [Ideas]
I'm using Inception as a backend for generating training data for building named entity recognition models with as purpose to anonymize and pseudonymize texts.
For the project I have a specific set of labels which I used in downstream processing to pseudonymize texts. It would be nice if I could create a project template specifically for that such that it can be shown next to these examples.

<img width="503" alt="image" src="https://github.com/user-attachments/assets/19fa8f83-0705-413d-9ec8-3c454fc8d9cc">

Would be nice if an exported zipped inception project could be linked to a template icon such that a user could click on the template icon and have already directly the setup which is specific for the project. Is that something which is already possible?

**Comment by reckart** [comment]
Currently, project templates need to be implemented programmatically. Many of the current templates also react differently depending on which feature flags are enabled in the system.

That said, having a functionality of using project exports as templates at this level is a good idea.

**Reply by jwijffels** [reply]
> Currently, project templates need to be implemented programmatically

Do you mean programmatically in the code base of inception or could they be created with e.g. pycaprio alongside dkpro-cassis?

**Reply by reckart** [reply]
In the code base, e.g. https://github.com/inception-project/inception/tree/main/inception/inception-project-initializers-phi/src/main/java/de/tudarmstadt/ukp/inception/project/initializers/phi

If you are into it, you could set yourself up a Maven project similar to e.g. `inception-project-initializers-phi`, build it and provide it as an [extra JAR](https://inception-project.github.io/releases/34.2/docs/admin-guide.html#_loading_extra_java_libraries) when you start INCEpTION.

**Reply by jwijffels** [reply]
thanks! I'll try it.
